modules/filters/mod_filter.c

```diff
@@ -176,6 +178,15 @@ static int filter_lookup(ap_filter_t *f, ap_filter_rec_t *filter)
                     match = 1;
                     break;
                 }
+                else if (sep                                     
+                    && strncmp(*type, r->content_type, sep) == 0     
+                    && strncmp(*type + sep, wildcard, strlen(wildcard)) == 0
+                    && (*type)[sep + strlen(wildcard)] == '\0') {                
```

Comment:
IDK if it compiles away, but I think the 2nd strncmp could just be strcmp allowing you to drop the last condition (and both strlens)